components/layout/dom.rs

```diff
@@ -140,6 +140,22 @@ impl LayoutBox {
         }
     }
 
+    pub(crate) fn with_base_first<T>(
```

Comment:
`with_base_first` sounds a bit odd because first then modifies `with`. When `first` is between `with` and `base` it modifies `base`. I have no idea why English works like that.

```suggestion
    pub(crate) fn with_first_base<T>(
```